.github/workflows/kind_e2e.yaml

```diff
@@ -0,0 +1,124 @@
+# ------------------------------------------------------------
+# Copyright (c) Microsoft Corporation and Dapr Contributors.
+# Licensed under the MIT License.
+# ------------------------------------------------------------
+
+name: E2E tests for K8s (KinD)
+
+on:
+  pull_request:
+    branches:
+      - master
+      - 'release-*'
+
+defaults:
+  run:
+    shell: bash
+    working-directory: ./src/github.com/dapr/cli
+
+jobs:
+  # This workflow runs our e2e tests in a local KinD cluster. Since it
+  # does not required a paid cluster and special credentials, it does
+  # not require /ok-to-test. That allows contributors to run E2E tests
+  # in their Draft PRs without explicit approval from Dapr
+  # maintainers.
+  #
+  # One other benefit of a fully-localized workflow is that tests are
+  # running on a fresh cluster everytime, thus being decoupled from
+  # shared resource issues.
+  #
+  # However, KinD currently does not support Windows nodes so this is
+  # not intended to be a complete replacement for our AKS-based E2E
+  # workflows.
+  e2e:
+    name: e2e
+    runs-on: ubuntu-latest
+    env:
+      GOVER: 1.16
+    strategy:
+      fail-fast: false # Keep running if one leg fails.
+      matrix:
+        k8s-version:
+        - v1.17.11
+        - v1.18.8
+        - v1.19.1
+        mode:
+        - ha
+        - non-ha
+        # Map between K8s and KinD versions.
+        # This is attempting to make it a bit clearer what's being tested.
+        # See: https://github.com/kubernetes-sigs/kind/releases/tag/v0.9.0
+        include:
+        - k8s-version: v1.17.11
```

Comment:
I think it is OK to have E2E test for multiple versions. This is good in case a future change unintentionally breaks compatibility with an older version.